assets/src/components/CustomLink/CustomLink.js

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@wordpress/element';
+import {URLInput} from '@wordpress/block-editor'
+import {TextControl as BaseTextControl} from '@wordpress/components';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+
+const TextControl = withCharacterCounter( BaseTextControl );
+
+export class CustomLink extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+
+  isCustomLinkValid() {
+    return !!this.props.value ? this.props.value.substr(0, 8).toLowerCase() === 'https://' : true;
+  }
+
+  render() {
+
+    const {__} = wp.i18n;
+    const allowSearch = !!this.props.allowSearch;
+
+    return (
+      <div>
+        { allowSearch ?
+          <URLInput
+            label={this.props.label}
+            className={this.props.className}
+            value={this.props.value}
+            onChange={this.props.onChange}
+            autoFocus={this.props.autoFocus}
+          />
+          :
+          <TextControl
+            label={this.props.label}
+            placeholder={this.props.placeholder}
+            value={this.props.value}
+            onChange={this.props.onChange}
+            disabled={this.props.disabled}
+            help={this.props.help}
+          />
+        }
+        {!this.isCustomLinkValid() &&
+        <span className='input_error'>{ __('The URL must start with "HTTPS://"', 'p4ge') }</span>}
```

Comment:
Maybe not within the scope of this ticket, but maybe we can add some kind of validation so that a post cannot be published when it has a block with an invalid url?